code.gs

```diff
@@ -2073,10 +2073,13 @@ function generateEmailNotification (forceDate) {
     [],
     settings.notifications.anticipateDays
       .map(function (days) {
+        // 24 * 60 * 60 * 1000 = 86400000
+        var thisNow = new Date(now.getTime() + days * 86400000);
```

Comment:
While troubleshooting all these problem with time and timezones I thought that we might have some troubles due to DST as well. More exactly if the script is set to be executed at midnight it will fail on the day DST ends.

```javascript
function testDate (year, month, day) {
  var date = new dateWithTimezone(year, month, day, 0, 0, 0, 'Europe/Rome');
  Logger.log('Date: ' + Utilities.formatDate(date, 'Europe/Rome', 'yyyy-MM-dd\'T\'HH:mm:ssXXX'));
  var date1 = new Date(date.getTime() + 24 * 60 * 60 * 1000);
  var date2 = new Date(date);
  date2.setDate(date.getDate() + 1);
  Logger.log('Adding ms: ' + Utilities.formatDate(date1, 'Europe/Rome', 'yyyy-MM-dd\'T\'HH:mm:ssXXX'));
  Logger.log('Adding days: ' + Utilities.formatDate(date2, 'Europe/Rome', 'yyyy-MM-dd\'T\'HH:mm:ssXXX'));
}

// 2018-08-01 is a "normal" day
testDate(2018, 7, 1);
// Date: 2018-08-01T00:00:00+02:00
// Adding ms: 2018-08-02T00:00:00+02:00
// Adding days: 2018-08-02T00:00:00+02:00

// DST ends on 2018-10-28 at 03:00 in Italy
testDate(2018, 9, 28);
// Date: 2018-10-28T00:00:00+02:00
// Adding ms: 2018-10-28T23:00:00+01:00     <-- WRONG DATE. This day only has 23 hours...
// Adding days: 2018-10-29T00:00:00+01:00
```
It's an extremely rare occurrence (1 in 365 days is affected and only if the user set execution time between 00:00 and 01:00 - actually that's not always true since not all countries DST is an hour long), but we should try to fix it anyway.  
I think using the `date2` method should prevent it from happening: if you agree I'll put it in together with the other fixes you suggested.

Reply:
Oh, yes - didn't even consider that. It's nice to catch even the edge-cases :-). Now that you've added it, of course `setDate(getDate() + 1)` looks much saner/more-robust to me too. It also pushes the slow microsecond-multiplying maths down inside the javascript-engine (v8), which will be optimized for bare-metal speed.